slick/slick.js

```diff
@@ -2198,16 +2198,22 @@
     Slick.prototype.setSlideClasses = function(index) {
 
         var _ = this,
-            centerOffset, allSlides, indexOffset, remainder;
+            centerOffset, allSlides, indexOffset, remainder, _slidesToShow;
+        var _slider = _.$slider.get()[0],
+            _slides = _.$slides.get();
 
-        allSlides = _.$slider
-            .find('.slick-slide')
-            .removeClass('slick-active slick-center slick-current')
-            .attr('aria-hidden', 'true');
+        allSlides = _slider.querySelectorAll('.slick-slide');
```

Comment:
Proposal:
`[].slice.call(_slider.querySelectorAll('.slick-slide'))`

It's set allSlides as an array and let you use directly `slice` or `forEach` below